archai/discrete_search/evaluators/progressive_training.py

```diff
@@ -26,24 +27,43 @@ def _stateful_training_fn(
     return _stateful_training_fn
 
 
-class ProgressiveTraining(ModelEvaluator):
+class ProgressiveTraining(ModelEvaluator, ProgressiveModelEvaluator):
     """Progressive training evaluator."""
 
-    def __init__(self, search_space: DiscreteSearchSpace, training_fn: Callable) -> None:
+    def __init__(self, training_fn: Callable, output_dir: str) -> None:
         """Initialize the evaluator.
 
         Args:
-            search_space: Search space.
             training_fn: Training function.
 
         """
 
-        self.search_space = search_space
         self.training_fn = training_fn
+        self.output_dir = Path(output_dir)
+
+        assert not self.output_dir.is_file(), f'Output directory {self.output_dir} is a file'
+        self.output_dir.mkdir(parents=True, exist_ok=True)
 
         # Training state buffer (e.g optimizer state) for each architecture id
         self.training_states = {}
 
+    @overrides
+    def save_checkpoint(self) -> str:
+        for archid, tr_state in self.training_states.items():
+            torch.save(tr_state, self.output_dir / f'{archid}.pt')
+        
+        return str(self.output_dir)
+
+    @overrides
+    def load_checkpoint(self, checkpoint: str) -> None:
+        checkpoint_dir = Path(checkpoint)
+        assert checkpoint_dir.is_dir(), f'Checkpoint {checkpoint} is not a directory'
+
+        for checkpoint_file in checkpoint_dir.glob('*.pt'):
+            archid = checkpoint_file.stem
+            tr_state = torch.load(checkpoint_file)
+            self.training_states[archid] = tr_state        
+
```

Comment:
Hmm this looks like it will accumulate more and more checkpoints, do you really want to load them all?  Or should you write a json file to the output folder containing the ones last saved by save_checkpoint and return that file name so that load_checkpoint only loads the most recent checkpoints that were saved?